### PR TITLE
Generalize equality and arithmetic operations for PolynomialTensor to allow differing keys

### LINKED ISSUE #274: Generalize __add__ and __eq__ methods of PolynomialTensor
As discussed in #240, we should be able to add and compare PolynomialTensors with different sets of keys (if the key is not present then it counts as the zero tensor). Currently, attempting these operations raises an error.

### PR DESCRIPTION
**Pull request by singular-value**
Resolves #274 and #301 .

### CODE REVIEW

**Comment by singular-value** [comment]
I signed it!

**Comment by kevinsung** [comment]
Thanks for the great PR! Ideally you would have tested this using EqualsTester (it would have caught the `__hash__` thing) but I won't require it. The tests for this class are kind of messy so that would be something of a hassle.

@babbush something is definitely up with Coveralls because it says "47 of 47 new or added lines in 2 files covered." I think after the one comment I made is addressed this should be good to go.

**Comment by babbush** [comment]
@kevinsung I agree. I am not sure what the problem is. Let me know once you feel the PR is good to go and I will override coveralls if necessary.

File: src/openfermion/ops/_polynomial_tensor.py [hunk 1]
```diff
@@ -173,22 +173,34 @@ def __setitem__(self, args, value):
...
             diff = max(diff, discrepancy)
+
         return diff < EQ_TOLERANCE
 
-    def __neq__(self, other_operator):
-        return not (self == other_operator)
+    def __ne__(self, other):
+        return not (self == other)
+
+    def __hash__(self):
```
Comment (kevinsung) [inline src/openfermion/ops/_polynomial_tensor.py]: Sorry, I should have been more clear. It's `__hash__ = None`.

**Comment by babbush** [comment]
coveralls appears to have gotten its act together.

**Comment by singular-value** [comment]
Awesome!

**Pull request merged**